### DISCUSSION TITLE
Update PiVPN without uninstall and reinstall?

### DISCUSSION THREAD

**Discussion by HeisenBug1** [pivpn/pivpn#1631] [Q&A] [answered]
Is there a way to update PiVPN without uninstalling and then reinstalling?

My PiVPN is 2 years old. I remember using the update script provided and something messed up. I also remember devs not allowing the use of the update script for that reason later on.

So, has the situation changed? Is there support for updating without reinstall in the latest 4.x version?

Thanks..

**Comment by coolapso** [comment, marked as answer]
Hi, May I ask why you want to update it?  are you missing any feature or bug fix in the management scripts?  there is very little reason to update PiVPN. 

What is important to keep up to date, is the VPN protocols and those you update through regular `sudo apt update && sudo apt upgrade`, https://docs.pivpn.io/install/#updating-pivpn

some more details here: https://github.com/pivpn/pivpn/wiki/Upgrading#how-do-we-upgrade-pivpn 

The pivpn update command has been disabled because in the past we introduced breaking changes and we have no way to "control" it. 
Versioning was introduced very recently with it I plan to ad back the update command but for sure, it won't update installations older than v4.x.x

**Reply by HeisenBug1** [reply]
It's working perfectly fine (2+ year old PiVPN). But all the hard work developers are putting in the latest version, I want to use it that's all. But to use it, i would have to uninstall, reconfigure and redistribute all the new keys again. So i was just wondering if the latest has any features to not require a full reinstall/reconfigure.. thanks..

**Reply by coolapso** [reply]
well that is great then! if you haven't noticed anything most likely you don't need any of the changes that were implemented in the meantime!  as I said, most of the changes we do go into the installer and very few go to the pivpn scripts.  the most important thing is that you keep the vpn protocol and regular updates on the OS level.

**Reply by HeisenBug1** [reply]
Okay makes sense.. thanks for the info..
